apps/builder/app/shared/copy-paste/plugin-markdown.ts

```diff
@@ -207,13 +208,19 @@ export const parse = (clipboardData: string, options?: Options) => {
 
 export const onPaste = (clipboardData: string) => {
   const data = parse(clipboardData);
-  if (data === undefined) {
+  const selectedPage = selectedPageStore.get();
+  if (data === undefined || selectedPage === undefined) {
     return;
   }
-
+  // paste to the root if nothing is selected
+  const instanceAddress = selectedInstanceAddressStore.get() ?? [
+    selectedPage.rootInstanceId,
+  ];
+  const [targetInstanceId] = instanceAddress;
   const dropTarget = findClosestDroppableTarget(
     instancesIndexStore.get(),
-    selectedInstanceIdStore.get()
+    // @todo accept instance address
```

Comment:
what will happen when pasting into slot rn?

Reply:
Nothing really, though will be easier to use already traversed list of ancestors